core/unix/signal.c

```diff
@@ -2709,12 +2715,13 @@ thread_set_self_context(void *cxt)
     asm("mov  %0, %%" ASM_XCX : : "m"(asm_jmp_tgt));
     asm("jmp  *%" ASM_XCX);
 #    endif /* MACOS/LINUX */
-#elif defined(AARCH64)
-    ASSERT_NOT_IMPLEMENTED(false); /* FIXME i#1569 */
+#elif defined(AARCH64) && !defined(DR_HOST_NOT_TARGET)
```

Comment:
This additional host check should not just be added to one of these arches: the behavior should be the same for other host-target combinations.  Instead, in general either A) DR_HOST_<arch> should be used as the condition for each of asm block or B) We don't support host!=target here since we only support non-code-execution use (decoding; standalone lib) and there should be a check for that with ASSERT_NOT_REACHED just like elsewhere in this file (e.g., notify_and_jmp_without_stack()).

Reply:
There's an `ASSERT_NOT_REACHED` after this ladder, but +1 on making it clearer like `notify_and_jmp_without_stack`.

Reply:
Done.